src/styles/index.scss

```diff
@@ -1,3 +1,84 @@
+$gap: 1px;
+$sizeSquare: 100px;
+$padding: 10px;
+$widthCalendar: $sizeSquare * 7 + $gap * 7 + $padding * 2;
+$days:
+  'mon' 0,
+  'tue' $sizeSquare + $gap,
+  'wen' 2 * ($sizeSquare + $gap),
+  'thu' 3 * ($sizeSquare + $gap),
+  'fri' 4 * ($sizeSquare + $gap),
+  'sat' 5 * ($sizeSquare + $gap),
+  'sun' 6 * ($sizeSquare + $gap);
+
+* {
+  box-sizing: border-box;
+}
+
 body {
   margin: 0;
 }
+
+.container {
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  height: 100vh;
+}
+
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  gap: 1px;
+  width: $widthCalendar;
+  padding: $padding;
+
+  &--first-day-sun {
+    @each $day, $margin in $days {
+      & :first-child {
+        margin-left: $margin;
+      }
+    }
+  }
+
+  &--length-month-30 {
+    @for $count from 30 through 30 {
```

Comment:
The `@for` loop in `&--length-month-30` is redundant since it only iterates once with `$count` being 30. You can directly use the selector `:nth-child(n + 31)` to hide the days beyond the 30th.